src/hooks/useScanApi.ts

```diff
@@ -1,73 +1,448 @@
+  const { t } = useTranslation();
+
+  const apiCall = useCallback(
+    (reportId: number): Promise<AxiosResponse<Blob>> =>
+      axios({
+        url: `${process.env.REACT_APP_REPORTS_SERVICE}${reportId}/`,
+        method: 'GET',
+        responseType: 'blob'
+      }),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (response: AxiosResponse<Blob>, reportId: number) => {
+      onAddAlert({
+        title: t('toast-notifications.description', {
+          context: 'report_downloaded',
+          name: reportId
+        }),
+        variant: 'success'
+      });
+
+      helpers.downloadData(
+        response.data,
+        `report_id_${reportId}_${new Date()
+          .toISOString()
+          .replace('T', '_')
+          .replace(/[^\d_]/g, '')}.tar.gz`,
+        'application/gzip'
+      );
+      return response;
+    },
+    [onAddAlert, t]
+  );
+
+  const callbackError = useCallback(
+    (error: AxiosError<ApiScanErrorType>, reportId: number) => {
+      onAddAlert({
+        title: t('toast-notifications.description', {
+          context: 'report_download_error',
+          name: reportId,
+          message: apiHelpers.extractErrorMessage(error?.response?.data)
+        }),
+        variant: 'danger'
+      });
+      return;
+    },
+    [onAddAlert, t]
+  );
+
+  const downloadReport = useCallback(
+    async (reportId: number) => {
+      let response;
+      try {
+        response = await apiCall(reportId);
+        return callbackSuccess(response, reportId);
+      } catch (error) {
+        if (axios.isAxiosError(error)) {
+          callbackError(error, reportId);
+        }
+        if (!helpers.TEST_MODE) {
+          console.error(error);
+        }
+        return undefined;
+      }
+    },
+    [apiCall, callbackSuccess, callbackError]
+  );
 
   return {
-    deleteScan,
-    onDeleteSelectedScans,
-    runScan,
-    pendingDeleteScan,
-    setPendingDeleteScan,
-    scanSelected,
-    setScanSelected,
-    getScanJobs,
+    apiCall,
+    callbackSuccess,
+    callbackError,
     downloadReport
   };
 };
-export default useScanApi;
+
+const useShowConnectionsApi = (setConnectionsData: (data: any) => void) => {
+  const apiCall = useCallback(
+    (source: SourceType): Promise<AxiosResponse> =>
+      axios.get(
+        `${process.env.REACT_APP_SCAN_JOBS_SERVICE}${source.connection.id}/connection/?page=1&page_size=1000&ordering=name&source_type=${source.id}`
+      ),
+    []
+  );
+
+  const callbackSuccess = useCallback(
+    (response: AxiosResponse) => {
+      setConnectionsData({
+        successful: response.data.results.filter((c: { status: string }) => c.status === 'success'),
+        failure: response.data.results.filter((c: { status: string }) => c.status === 'failure'),
+        unreachable: response.data.results.filter((c: { status: string }) => !['success', 'failure'].includes(c.status))
+      });
+      return;
+    },
+    [setConnectionsData]
+  );
```

Comment:
appears I forgot to mention something. It's an addition to the `const callbackError`... you're going to come across this if/when you attempt to distinguish between the `success` and `error` callbacks 

Because we're trapping the API calls... and for the most part we've just been using `finally` so this hasn't been needed... in order to leverage the error callback we must return `Promise.reject(...` inside of
```
const callbackError = useCallback((error: AxiosError) => Promise.reject(error),[]);
```

In order for it to be consumed like

```
yourApiCallWrapper.then(success => console.log(success), theErrorFromPromiseReject => console.log(theErrorFromPromiseReject));
```
or with a `catch`
```
yourApiCallWrapper.catch(theErrorFromPromiseReject => console.log(theErrorFromPromiseReject));
```